packages/shared/decorators/with-loader.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from "react";
+
+const withLoader = <T extends { [k: string]: any }>(
+  Component: React.ComponentType<T>
+): React.ComponentType<WithLoaderProps & T> => props => {
+  const { loader, condition = true } = props;
+  if (condition) return <Component {...props} />;
+  else if (loader) return loader;
+  return null;
+};
+
+export interface WithLoaderProps {
+  loader?: JSX.Element;
+  condition?: boolean;
```

Comment:
I suggest using reverse property `isLoading` that is `false` by default.

Reply:
Do you mean rename `condition` prop to this name?
This is correct for case when decorator is used with loader, but if you use it to show or hide a component prop doesn't mean that data is loading.